narayana/ArjunaJTA/jta/tests/classes/io/narayana/perf/product/ProductWorker.java

```diff
@@ -21,71 +21,43 @@
  */
 package io.narayana.perf.product;
 
-import com.arjuna.ats.jta.xa.performance.XAResourceImpl;
-
-import javax.transaction.*;
+import javax.transaction.HeuristicMixedException;
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.NotSupportedException;
+import javax.transaction.RollbackException;
+import javax.transaction.SystemException;
+import javax.transaction.TransactionManager;
 import javax.transaction.xa.XAResource;
-import java.sql.Connection;
-import java.sql.SQLException;
-import java.sql.Statement;
+
+import com.arjuna.ats.jta.xa.performance.XAResourceImpl;
 
 public class ProductWorker<Void> {
 
     ProductInterface prod;
+    
+    protected XAResource xaResource1 = new XAResourceImpl();
+    protected XAResource xaResource2 = new XAResourceImpl();
+
+    private TransactionManager ut;
 
     public ProductWorker(ProductInterface prod) {
         this.prod = prod;
     }
 
-    public XAResource getXAResource() {
-        return new XAResourceImpl();
-    }
-
     public void doWork() throws SystemException, NotSupportedException, RollbackException, HeuristicRollbackException, HeuristicMixedException {
-        TransactionManager ut = prod.getTransactionManager();
-
         ut.begin();
-        ut.getTransaction().enlistResource(getXAResource());
-        ut.getTransaction().enlistResource(getXAResource());
```

Comment:
Bitronix needs to override getXAResource() since you cannot enlist an arbitrary XAResource